### PR TITLE
Reversing a linked list c++

### LINKED ISSUE #193: Reversing a linked list
Would like to add an algorithm for reversing a given linked list in C++. Please assign this to me under gssoc2021.

### PR DESCRIPTION
**Pull request by dhruvinfo28**
closes #193 . Algorithm for reversing a linked list added. Space and time complexity mentioned and i/o given

### CODE REVIEW

**Comment by dhruvinfo28** [comment]
@raksha009 Kindly review please

File: Code/C++/reverse_a_linked_list.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#include<bits/stdc++.h>
```
Comment (rudrakshi99) [inline Code/C++/reverse_a_linked_list.cpp:1]: Add a brief description of the algorithm.

**Review by rudrakshi99 — CHANGES_REQUESTED** [review]

**Comment by dhruvinfo28** [comment]
@rudrakshi99  changes done . Kindly review

File: Code/C++/reverse_a_linked_list.cpp [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+    3. We traverse the linked list until curr becomes null.
+    4. Make ahead = curr->next
+    5. Now make curr's pointer point to prev i.e. curr->next = prev
+    6. Move prev one step further i.e. prev = curr
+    7. Move curr one step further i.e curr = ahead
+    8. After the loop terminates, prev will be pointing to the new head i.e the last node in the list. So make head = prev
+*/
 #include<bits/stdc++.h>
 using namespace std;
 struct Node{
```
Comment (rudrakshi99) [inline Code/C++/reverse_a_linked_list.cpp:15]: use `class` instead of `struct`

**Review by rudrakshi99 — CHANGES_REQUESTED** [review]

File: Code/C++/reverse_a_linked_list.cpp (hunk 2 above)
Comment (dhruvinfo28) [reply Code/C++/reverse_a_linked_list.cpp]: @rudrakshi99 Requested changes done

**Review by rudrakshi99 — APPROVED** [review]

File: Code/C++/reverse_a_linked_list.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+8
+10 20 30 40 50 60 70 80
+
+Output:
+Your list: 
+80 70 60 50 40 30 20 10 
+Reversed list 
+10 20 30 40 50 60 70 80
+
+Time Complexity : O(n)
```
Comment (raksha009) [inline Code/C++/reverse_a_linked_list.cpp:108]: Specify n here.

**Review by raksha009 — CHANGES_REQUESTED** [review]
Plz add some comments in your code explaining its working.

File: Code/C++/reverse_a_linked_list.cpp (hunk 3 above)
Comment (dhruvinfo28) [reply Code/C++/reverse_a_linked_list.cpp]: Made the requested changes @raksha009 Kindly review

**Review by raksha009 — APPROVED** [review]
Great Work!

**Pull request merged**